velox/exec/HashTable.h

```diff
@@ -258,9 +258,9 @@ class BaseHashTable {
   /// owned by 'this'.
   virtual int64_t allocatedBytes() const = 0;
 
-  /// Deletes any content of 'this' but does not free the memory. Can
-  /// be used for flushing a partial group by, for example.
-  virtual void clear() = 0;
+  /// Deletes any content of 'this'. If 'freeTable' is false, then hash table is
```

Comment:
Why do we need this flag? What is freeTable == true for? Why can't we just destroy the whole HashTable object?

Reply:
This flag just helps to free all the memory resources. If we need to destroy the whole hash table object, then there will be more changes to the relevant code path in hash probe as the code path need to handle a null hash table instead of an empty one. Thanks!